### PR TITLE
Rename "grypedb" feed to be called "vulnerabilities".

### PR DESCRIPTION
**Pull request by Vijay-P**
**What this PR does / why we need it**:
This PR adds feed display mappers to policy engine VulnerabilityProvider. These mappers will allow us to rename feeds at ingress(API) and egress(API response). I've purposefully tried to constrain my changes as much as possible to the API layer of the code. If you are wondering why I did not follow the approach of changing the name from the origin of the data throughout the system, the answer is because the policy engine expects parity between configuration, feed source, and the processing of the feed by name (i.e. the name of the feed should match throughout the system). This model is no longer suitable for the future plans for the policy engine, and further work is required to divorce the policy engine from that model. If you are curious about the exploratory work that I did to attempt to implement this change as described above, you can find that here: https://github.com/anchore/anchore-engine/pull/1188

**Which issue this PR fixes** *(optional, in `fixes #<issue number>)(, fixes #<issue_number, ...)` format, will close the issue when PR is merged*: fixes #:

**Special notes**:

### CODE REVIEW

**Review by zburstein — APPROVED** [review]
All the changes look good to me! Do we have test coverage for toggle_group_enabled, delete_feed, and delete_group? Im not seeing any, but not sure if I am maybe missing something. Its something that might be worth adding since those endpoints have been changed

**Comment by Vijay-P** [comment]
> All the changes look good to me! Do we have test coverage for toggle_group_enabled, delete_feed, and delete_group? Im not seeing any, but not sure if I am maybe missing something. Its something that might be worth adding since those endpoints have been changed

I don't think we have any functional tests for those actually.

File: anchore_engine/services/policy_engine/api/controllers/feeds.py [hunk 1]
```diff
@@ -154,14 +165,18 @@ def toggle_feed_enabled(feed, enabled):
...
-            message="Enabling and disabling groups for grypedb feed is not currently supported.",
+            message="Enabling and disabling groups for {} feed with the grype vulnerability provider enabled is not currently supported.".format(
+                feed
+            ),
             detail={},
         )
     session = db.get_session()
     try:
-        g = db.set_feed_group_enabled(session, feed, group, enabled)
+        g = db.set_feed_group_enabled(session, internal_feed_name, group, enabled)
```
Comment (dspalmer99) [inline anchore_engine/services/policy_engine/api/controllers/feeds.py:179]: I know this was already there, but can you rename the single-character variable?

File: anchore_engine/services/policy_engine/api/controllers/feeds.py [hunk 2]
```diff
@@ -182,14 +197,18 @@ def toggle_group_enabled(feed, group, enabled):
 
 @authorizer.requires_account(with_types=INTERNAL_SERVICE_ALLOWED)
 def delete_feed(feed):
-    if feed == GRYPE_DB_FEED_NAME:
+    provider = get_vulnerabilities_provider()
+    internal_feed_name = provider.display_mapper.get_internal_name(feed)
+    if isinstance(provider, GrypeProvider) and internal_feed_name == GRYPE_DB_FEED_NAME:
         raise HTTPNotImplementedError(
-            message="Deleting the grypedb feed is not yet supported.",
+            message="Deleting the {} feed with the grype vulnerability provider enabled is not yet supported.".format(
```
Comment (dspalmer99) [inline anchore_engine/services/policy_engine/api/controllers/feeds.py:204]: Can you use the safer string format option, e.g.: [https://github.com/anchore/anchore-engine/blob/6ca1795bf20114a173ab348d8184c2e647c987d1/anchore_engine/clients/grype_wrapper.py#L309](https://github.com/anchore/anchore-engine/blob/6ca1795bf20114a173ab348d8184c2e647c987d1/anchore_engine/clients/grype_wrapper.py#L309)

File: anchore_engine/services/policy_engine/api/controllers/feeds.py [hunk 3]
```diff
@@ -182,14 +197,18 @@ def toggle_group_enabled(feed, group, enabled):
...
+                feed
+            ),
             detail={},
         )
     session = db.get_session()
     try:
-        f = db.lookup_feed(db_session=session, feed_name=feed)
+        f = db.lookup_feed(db_session=session, feed_name=internal_feed_name)
```
Comment (dspalmer99) [inline anchore_engine/services/policy_engine/api/controllers/feeds.py:211]: Same, please rename the single character variable.

File: anchore_engine/services/policy_engine/api/controllers/feeds.py [hunk 4]
```diff
@@ -205,7 +224,7 @@ def delete_feed(feed):
         session.rollback()
 
     try:
-        f = sync.DataFeeds.delete_feed(feed)
+        f = sync.DataFeeds.delete_feed(internal_feed_name)
```
Comment (dspalmer99) [inline anchore_engine/services/policy_engine/api/controllers/feeds.py:227]: Ditto (although I know you didn't add this(.

File: anchore_engine/services/policy_engine/api/controllers/feeds.py [hunk 5]
```diff
@@ -220,14 +239,20 @@ def delete_feed(feed):
 
 @authorizer.requires_account(with_types=INTERNAL_SERVICE_ALLOWED)
 def delete_group(feed, group):
-    if feed == GRYPE_DB_FEED_NAME:
+    provider = get_vulnerabilities_provider()
+    internal_feed_name = provider.display_mapper.get_internal_name(feed)
+    if isinstance(provider, GrypeProvider) and internal_feed_name == GRYPE_DB_FEED_NAME:
         raise HTTPNotImplementedError(
-            message="Deleting individual groups for the grypedb feed is not yet supported.",
+            message="Deleting individual groups for the {} feed with the grype vulnerability provider enabled is not yet supported.".format(
```
Comment (dspalmer99) [inline anchore_engine/services/policy_engine/api/controllers/feeds.py:246]: Same comment as above re string formatting.

**Review by dspalmer99 — CHANGES_REQUESTED** [review]
A few small changes, but overall it looks good.

File: anchore_engine/services/policy_engine/api/controllers/feeds.py (hunk 3 above)
Comment (Vijay-P) [reply anchore_engine/services/policy_engine/api/controllers/feeds.py]: done

File: anchore_engine/services/policy_engine/api/controllers/feeds.py (hunk 4 above)
Comment (Vijay-P) [reply anchore_engine/services/policy_engine/api/controllers/feeds.py]: done

File: anchore_engine/services/policy_engine/api/controllers/feeds.py (hunk 1 above)
Comment (Vijay-P) [reply anchore_engine/services/policy_engine/api/controllers/feeds.py]: done

File: anchore_engine/services/policy_engine/api/controllers/feeds.py (hunk 5 above)
Comment (Vijay-P) [reply anchore_engine/services/policy_engine/api/controllers/feeds.py]: switched to f-strings

File: anchore_engine/services/policy_engine/api/controllers/feeds.py (hunk 2 above)
Comment (Vijay-P) [reply anchore_engine/services/policy_engine/api/controllers/feeds.py]: switched to f-strings

**Comment by Vijay-P** [comment]
> All the changes look good to me! Do we have test coverage for toggle_group_enabled, delete_feed, and delete_group? Im not seeing any, but not sure if I am maybe missing something. Its something that might be worth adding since those endpoints have been changed

I'll make an issue for creating tests for these and follow up in subsequent PR.

**Review by dspalmer99 — APPROVED** [review]

**Pull request merged**